src/shared/getNotifyClient.ts

```diff
@@ -0,0 +1,33 @@
+import { NotifyClient } from "notifications-node-client";
+import * as config from "server/config";
+import { get } from "lodash";
+
+export function getNotifyClient() {
+  // TODO: Use NotifyClient type instead of any
+  let notifyClient: any;
```

Comment:
was this always scoped to the function? was it changed recently? essentially the notify client is being initialised each time we want to send an email. 

easiest thing to do for now is just move let notifyClient outside of the function (singleton pattern). the code `if (notifyClient === undefined)` checks if a notifyClient exists, and if not, initialises it (lazy initialisation or lazy init) 

If `notifyClient` is scoped to the function, it means it will always be undefined and initialised each time. I'm pretty sure the code is expecting a singleton (it's expecting `getNotifyClient` to get the same instance of the client).

Reply:
yeah that makes sense 👍🏿

Reply:
I always thought the singleton pattern could only be used in a class. How does it work with a function? If a class is initialised outside of a function as a variable, is that saved in memory so when the function runs, it wouldn't initialised the class again?